_posts/2019-07-29-plonk-stack.md

```diff
@@ -0,0 +1,211 @@
+
+The PLONK stack combines the following Cloud Native technologies from the Cloud Native Landscape:
+
+* Prometheus
+* Linkerd
+* OpenFaaS
+* NATS
+* Kubernetes
+
+Out of the projects listed, four are CNCF-hosted. Of the CNFC projects, 2 are "graduated" and the other two are "incubating" and showing signs of graduation soon. In this post I'll explain what each of these CNCF projects lends to OpenFaaS, and how they can build a first-class FaaS or PaaS when combined.
+
+### The projects
+
+Let's take a quick look at each project, where you can get it from and how it is used.
+
+#### Prometheus
+
+> The Prometheus monitoring system and time series database.
+
+With its [origins in the engineering team at SoundCloud](https://www.youtube.com/watch?v=cdKc8ePbj4A), Prometheus is now the de-facto monitoring solution for Cloud Native projects. It combines a simple interface with a powerful query language to monitor and observe microservices and functions, which are the two primitives of any FaaS or PaaS.
+
+[Prometheus](https://prometheus.io) is the core project, but the ecosystem is rich, and growing.
+
+* [node_exporter](https://github.com/prometheus/node_exporter) - get metrics from machines in your cluster, VMs, or servers
+* [alertmanager](https://github.com/prometheus/alertmanager) - fire off alerts according to custom rules including integrations to Hipchat, Slack, Pagerduty, and more
+
+There are two ways to expose metrics to Prometheus:
+
+* add an instrumentation endpoint to each of your applications using a [client library](https://github.com/prometheus/client_golang)
+* create an "exporter" which leaves your code unmodified
+
+The project community encourage users to add instrumentation endpoints such as `/metrics` to their applications.
+
+Once you've added instrumentation endpoints, and decided what to record, you set up a Prometheus server and tell it which endpoints to start collecting from. Collecting is also called "scraping" and multiple "service-discovery" mechanisms exist such as DNS, or Consul which means no manual lists of endpoints are required. This makes Prometheus perfect for auto-scaling systems.
+
+Here's an example of the dashboard you can deploy with OpenFaaS using Grafana. Grafana is an open-source dashboarding tool which can render and save collections of queries onto the Prometheus time-series.
```

Comment:
I'd switch the sentences around here.  After the first i wondered why we were talking about grafana.